### ISSUE TITLE
`nix edit` is broken on master

### ISSUE THREAD

**Issue by thufschmitt** [NixOS/nix#9652]
**Describe the bug**

`nix edit` fails on master with `error: cannot open '<path to the Nix file>' in an editor because it has no physical path`.

The file does exist, and the same command on Nix 2.19 succeeds.

**Steps To Reproduce**

```console
$ nix run github:NixOS/nix/8e64cd59b0fe1a4e1d3ba8d359ea0f9af2936f94 -- edit nixpkgs#hello
error: cannot open '/nix/store/n0c1l1jpyvhmyp6lqpf5wrf4hnalpc72-source/pkgs/by-name/he/hello/package.nix' in an editor because it has no physical path
```

**Expected behavior**

Should open the relevant file

**`nix-env --version` output**

**Additional context**

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by thufschmitt** [comment]
@edolstra I suppose it's related to the recent accessor changes?

**Comment by Artturin** [comment]
Fixed by https://github.com/NixOS/nix/issues/9652

**Comment by eclairevoyant** [comment]
fix PR: #10814
